workflow-hotfix.md

```diff
@@ -0,0 +1,306 @@
+# Workflow for Creating a Hotfix Release
+
+Creating a hotfix for a release is very simple. There are basic steps
+to follow to create a hotfix release. As an example, I'll use the
+[OMS project][], creating a mythical hotfix solely for example
+purposes.
+
+The basic steps to creating a hotfix release are:
+
+1. [Check in your patches](#check-in-your-patches)
+2. [Set subproject references](#set-subproject-references)
+3. [Set up superproject](#set-up-superproject)
+4. [Build your project](#build-your-project)
+5. [Create tags and clean up](#create-tags-and-clean-up)
+
+All command examples assume you followed [Setting up git][] guidelines.
+
+-----
+
+### Check in your patches
+
+For reference, the [OMS project][] has five subprojects:
+
+1. dsc
+2. omi
+3. omsagent
+4. opsmgr
+5. pal
+
+First, you should probably be sure that your projects are completely
+up to date, and that you're set to the master branch for each
+subproject:
+
+```
+cd ~/dev/bld-omsagent
+git co-master
+git sync-subs
+```
+
+For our hotfix, let's assume we need to modify the OMS Agent
+configuration file, [omsagent.conf][]. First, we modify the file and
+check in the modified result into a new branch:
+
+```
+cd omsagent
+git checkout -b jeff-hotfix
+emacs installer/conf/omsagent.conf
+git commit -m "Fix parameters passed to omsadmin.sh for onboarding"
+```
+
+Obviously, other changes can be made to other files as part of this
+hotfix.
+
+### Set subproject references
+
+Now that we have a hotfix checked into the [omsagent.conf][] file, we
+need to set up other subprojects as desired. For purposes of discussion,
+let's say that we need to choose a prior version of [DSC][] than what is
+checked into mainline.
+
+We last left our shell cd'ed into the omsagent subproject. So, we'll
+issue a `git lol` command against DSC as follows:
+
+```
+cd ../dsc
+git lol
+```
+
+You'll see output such as:
+
+>```
+* a9ccdf5 (HEAD, origin/master, origin/dywu-nxOMSPlugin, origin/HEAD, master) Fix formatting and link in readme
+* 2c3cb16 Update readme.md
+* 6a41725 Update readme.md
+* 087b08f Correct Debian version format of OMI to x.y.z.r (rather than x.y.z-r)
+* d0e6421 (tag: v1.1.1-70) Fix up log rotation for OMI (consider moving to OMI project later)
+* b88c2f7 Explicitly disable -Werror in ./configure
+* 3b8679c Fix compile warnings in EngineHelper
+* 499a8a5 Add instructions to build from source
+>```
+
+Say we want to not ship our hotfix the first three changes, instead
```

Comment:
Rephrase to "Let's say we don't want to ship the three most recent commints in the hotfix"